### PR TITLE
fix: Fix dependencies

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR fixes the dependencies as suggested in https://github.com/getsentry/snuba-sdk/commit/b7a64da26d17987fd233351a4d37a883fd7fdd9d#r138544221.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install -r test-requirements.txt
+        pip install -e .
```
[the remark this answers is not in the document]
Comment (iambriccardo) [reply .github/workflows/tests.yml:52]: Which command should I use just `pip install -r -e .`?

File: setup.py [hunk 2]
```diff
@@ -37,7 +37,10 @@ def get_file_text(file_name: str) -> str:
     package_data={"snuba_sdk": ["py.typed"]},
     zip_safe=False,
     license="FSL-1.0-Apache-2.0",
-    install_requires=[],
+    install_requires=[
+        "parsimonious>=0.10.0",
+        "types-parsimonious>=0.10.0.8"
```
[the remark this answers is not in the document]
Comment (iambriccardo) [reply setup.py:42]: Should I put them into `requirements.txt`?

File: .github/workflows/tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (asottile-sentry) [reply .github/workflows/tests.yml:52]: `pip install -r test-requirements.txt -e .` for example -- and when you have multiple `-r` you can include them both in the same command like `pip install -r t1.txt -r t2.txt -r t3.txt`

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (asottile-sentry) [reply setup.py:42]: I think maybe linter-requirements.txt ? unclear exactly how this repo is set up

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+parsimonious>=0.10.0
```
Comment (asottile-sentry) [inline requirements.txt:1]: libraries should not have `requirements.txt` -- requirements.txt is for applications

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -48,7 +48,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install -r test-requirements.txt
+        pip install -r requirements.txt -r test-requirements.txt
```
Comment (asottile-sentry) [inline .github/workflows/tests.yml:51]: this is not correct -- this bypasses your library and does not test that it is packaged properly (the root cause of why this broke in the first place!)

the editable install which was here before is the right way to do this

**Review by asottile-sentry — APPROVED** [review]

**Review by evanh — APPROVED** [review]

**Pull request merged**